nyxt: 2.2.4 -> 3.1.0

dariof4 opened the pull request:
###### Description of changes

Update Nyxt from version 2.2.4 to version 3.1.0.
Release notes since 2.2.4:
https://nyxt.atlas.engineer/article/release-3.0.0.org
https://nyxt.atlas.engineer/article/release-3.1.0.org

Fixes:  #217888

This also updates the lisp dependencies needed to build Nyxt 3.1.0 and adds `prompter` which was transferred from Nyxt to an external library not yet on quicklisp, it also changes the build from `nyxt/gtk-application` to `nyxt/gi-gtk-application` since that seems to be the one to package according to Nyxt devs (the makefile and Guix package `gi-gtk`).

I have a few concerns though:

1. Each time Nyxt is launched it for some reason tries to recompile `nasdf` (which it fails since it can't create `/homeless-shelter`) and crashes with an `system nyxt out of date` error, I found several ways to "fix" this, either setting `CL_SOURCE_REGISTRY=$CL_SOURCE_REGISTRY:$(pwd)//` in `postConfigure` which makes it stop trying to recompile itself, using `(asdf:disable-output-translations)` like `stumpwm` does it, which makes it still try to recompile `nasdf` and fails, but makes Nyxt run fine, or setting `HOME=/tmp` which allows Nyxt to successfully recompile `nasdf` in `/tmp/.cache` (Guix sets `HOME=/tmp` for Nyxt as well, may be related?).
I'm a little bit out of my depth here and have no clue why Nyxt would be trying to recompile `nasdf`, it seems to be something to do with `CL_SOURCE_REGISTRY`, but I don't have enough experience with CL to understand this at all :sweat_smile:
I would appreciate if someone more experienced could shed some light as to what is the best solution. 

2. Should this PR be split up into multiple? and if so in how many should it be split? I tried to keep the commits be logically separate so this should be easy.
 
3. I think using `wrapLisp` within Nyxt's `default.nix` (needed to compile Nyxt with the `--dynamic-memory-size` flag to increase the heap size) causes quite a few extra builds, since the way I understand it can't use the cache for all the lisp dendencies of Nyxt, since it uses `wrapLisp`, it might be better to apply `--dynamic-memory-size` globally to `sbcl`?

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

dariof4 committed:
lisp-modules: add updated lispLibs for nyxt

Adds various updated lisp packages that are needed to update nyxt to 3.1.0.

dariof4 committed:
lisp-modules.prompter: init at 0.1.0

Promter was an internal part of nyxt that was split off as it's own
external library in nyxt 3.0.0.

dariof4 committed:
nyxt: 2.2.4 -> 3.1.0

`WEBKIT_FORCE_SANDBOX=0` is no longer needed since web extension
support has been diabled by default.
ref: https://github.com/atlas-engineer/nyxt/pull/2897

Release notes:
https://nyxt.atlas.engineer/article/release-3.0.0.org
https://github.com/atlas-engineer/nyxt/releases/tag/3.1.0

dariof4 commented:
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>46 packages failed to build:</summary>
  <ul>
    <li>sbclPackages._40ants-doc-test</li>
    <li>sbclPackages.april-demo_dot_ncurses</li>
    <li>sbclPackages.bus</li>
    <li>sbclPackages.cl-clblas-test</li>
    <li>sbclPackages.cl-feedparser</li>
    <li>sbclPackages.cl-feedparser-tests</li>
    <li>sbclPackages.cl-libpuzzle-test</li>
    <li>sbclPackages.cl-mecab-test</li>
    <li>sbclPackages.cl-oclapi-test</li>
    <li>sbclPackages.cl-pslib</li>
    <li>sbclPackages.cl-pslib-barcode</li>
    <li>sbclPackages.clack-handler-fcgi</li>
    <li>sbclPackages.clath</li>
    <li>sbclPackages.cletris-test</li>
    <li>sbclPackages.clim-widgets</li>
    <li>sbclPackages.climon-test</li>
    <li>sbclPackages.data-frame</li>
    <li>sbclPackages.dfio</li>
    <li>sbclPackages.dom</li>
    <li>sbclPackages.exscribe</li>
    <li>sbclPackages.gendl</li>
    <li>sbclPackages.geysr</li>
    <li>sbclPackages.graphs</li>
    <li>sbclPackages.gwl</li>
    <li>sbclPackages.gwl-graphics</li>
    <li>sbclPackages.hamcrest-tests</li>
    <li>sbclPackages.ledger</li>
    <li>sbclPackages.lisp-stat</li>
    <li>sbclPackages.maiden-dictionary</li>
    <li>sbclPackages.mockingbird-test</li>
    <li>sbclPackages.monomyth</li>
    <li>sbclPackages.nyxt</li>
    <li>sbclPackages.nyxt-ubuntu-package</li>
    <li>sbclPackages.pgloader</li>
    <li>sbclPackages.pkg-doc</li>
    <li>sbclPackages.plot</li>
    <li>sbclPackages.prometheus_dot_test_dot_all</li>
    <li>sbclPackages.regression</li>
    <li>sbclPackages.robot</li>
    <li>sbclPackages.ta2</li>
    <li>sbclPackages.tasty</li>
    <li>sbclPackages.translators</li>
    <li>sbclPackages.tree</li>
    <li>sbclPackages.wire-world</li>
    <li>sbclPackages.yadd</li>
    <li>sbclPackages.zaserve</li>
  </ul>
</details>
<details>
  <summary>374 packages built:</summary>
  <ul>
    <li>nyxt</li>
    <li>sbclPackages._40ants-doc-full</li>
    <li>sbclPackages.april</li>
    <li>sbclPackages.april-demo_dot_cnn</li>
    <li>sbclPackages.april-demo_dot_fnn</li>
    <li>sbclPackages.april-lib_dot_dfns_dot_array</li>
    <li>sbclPackages.april-lib_dot_dfns_dot_graph</li>
    <li>sbclPackages.april-lib_dot_dfns_dot_numeric</li>
    <li>sbclPackages.april-lib_dot_dfns_dot_power</li>
    <li>sbclPackages.april-lib_dot_dfns_dot_string</li>
    <li>sbclPackages.april-lib_dot_dfns_dot_tree</li>
    <li>sbclPackages.april-xt_dot_uzuki</li>
    <li>sbclPackages.assoc-utils-test</li>
    <li>sbclPackages.aws-foundation</li>
    <li>sbclPackages.aws-sdk</li>
    <li>sbclPackages.birch_dot_test</li>
    <li>sbclPackages.bit-smasher-test</li>
    <li>sbclPackages.bnf_dot_test</li>
    <li>sbclPackages.can-test</li>
    <li>sbclPackages.carrier</li>
    <li>sbclPackages.caveman2</li>
    <li>sbclPackages.caveman2-test</li>
    <li>sbclPackages.caveman2-widgets</li>
    <li>sbclPackages.caveman2-widgets-bootstrap</li>
    <li>sbclPackages.caveman2-widgets-bootstrap-test</li>
    <li>sbclPackages.caveman2-widgets-test</li>
    <li>sbclPackages.check-bnf_dot_test</li>
    <li>sbclPackages.chirp-dexador</li>
    <li>sbclPackages.ciao</li>
    <li>sbclPackages.circular-streams-test</li>
    <li>sbclPackages.cl-amqp_dot_test</li>
    <li>sbclPackages.cl-annot-prove</li>
    <li>sbclPackages.cl-annot-prove-test</li>
    <li>sbclPackages.cl-ansi-text</li>
    <li>sbclPackages.cl-ansi-text_dot_test</li>
    <li>sbclPackages.cl-ascii-art</li>
    <li>sbclPackages.cl-association-rules-tests</li>
    <li>sbclPackages.cl-base58-test</li>
    <li>sbclPackages.cl-bson-test</li>
    <li>sbclPackages.cl-cache-tables-tests</li>
    <li>sbclPackages.cl-cognito</li>
    <li>sbclPackages.cl-coinpayments</li>
    <li>sbclPackages.cl-colors2</li>
    <li>sbclPackages.cl-cookie</li>
    <li>sbclPackages.cl-cookie-test</li>
    <li>sbclPackages.cl-coroutine-test</li>
    <li>sbclPackages.cl-coveralls</li>
    <li>sbclPackages.cl-coveralls-test</li>
    <li>sbclPackages.cl-covid19</li>
    <li>sbclPackages.cl-darksky</li>
    <li>sbclPackages.cl-darksky-test</li>
    <li>sbclPackages.cl-data-structures-tests</li>
    <li>sbclPackages.cl-debug-print-test</li>
    <li>sbclPackages.cl-disque-test</li>
    <li>sbclPackages.cl-dotenv-test</li>
    <li>sbclPackages.cl-emoji-test</li>
    <li>sbclPackages.cl-events_dot_test</li>
    <li>sbclPackages.cl-fixtures-test</li>
    <li>sbclPackages.cl-gearman-test</li>
    <li>sbclPackages.cl-gists</li>
    <li>sbclPackages.cl-gists-test</li>
    <li>sbclPackages.cl-gopher</li>
    <li>sbclPackages.cl-haml-test</li>
    <li>sbclPackages.cl-hash-table-destructuring-test</li>
    <li>sbclPackages.cl-incognia</li>
    <li>sbclPackages.cl-ini-test</li>
    <li>sbclPackages.cl-intbytes-test</li>
    <li>sbclPackages.cl-jsx-test</li>
    <li>sbclPackages.cl-kraken</li>
    <li>sbclPackages.cl-ksuid</li>
    <li>sbclPackages.cl-ksuid-test</li>
    <li>sbclPackages.cl-libsvm-format-test</li>
    <li>sbclPackages.cl-locale-test</li>
    <li>sbclPackages.cl-ltsv-test</li>
    <li>sbclPackages.cl-markup-test</li>
    <li>sbclPackages.cl-mustache-test</li>
    <li>sbclPackages.cl-notebook</li>
    <li>sbclPackages.cl-online-learning-test</li>
    <li>sbclPackages.cl-ply-test</li>
    <li>sbclPackages.cl-project</li>
    <li>sbclPackages.cl-project-test</li>
    <li>sbclPackages.cl-punch-test</li>
    <li>sbclPackages.cl-random-forest-test</li>
    <li>sbclPackages.cl-reexport-test</li>
    <li>sbclPackages.cl-rules-test</li>
    <li>sbclPackages.cl-selenium</li>
    <li>sbclPackages.cl-selenium-test</li>
    <li>sbclPackages.cl-ses4</li>
    <li>sbclPackages.cl-shellwords-test</li>
    <li>sbclPackages.cl-slug-test</li>
    <li>sbclPackages.cl-ssdb-test</li>
    <li>sbclPackages.cl-statsd_dot_test</li>
    <li>sbclPackages.cl-strings-tests</li>
    <li>sbclPackages.cl-tasukete</li>
    <li>sbclPackages.cl-tasukete-test</li>
    <li>sbclPackages.cl-telebot</li>
    <li>sbclPackages.cl-telegram-bot</li>
    <li>sbclPackages.cl-test-more</li>
    <li>sbclPackages.cl-textmagic</li>
    <li>sbclPackages.cl-textmagic-test</li>
    <li>sbclPackages.cl-toml-test</li>
    <li>sbclPackages.cl-transmission-test</li>
    <li>sbclPackages.cl-voxelize-test</li>
    <li>sbclPackages.cl-weather-jp</li>
    <li>sbclPackages.cl-weather-jp-test</li>
    <li>sbclPackages.cl-webdriver-client</li>
    <li>sbclPackages.cl-webdriver-client-test</li>
    <li>sbclPackages.cl-webkit2</li>
    <li>sbclPackages.cl-yaclyaml</li>
    <li>sbclPackages.cl-yaclyaml-tests</li>
    <li>sbclPackages.cl-zipper-test</li>
    <li>sbclPackages.cl_plus_ssl_dot_test</li>
    <li>sbclPackages.clache-test</li>
    <li>sbclPackages.clack-handler-woo</li>
    <li>sbclPackages.clack-handler-wookie</li>
    <li>sbclPackages.clack-pretend</li>
    <li>sbclPackages.clack-static-asset-middleware-test</li>
    <li>sbclPackages.clack-test</li>
    <li>sbclPackages.claxy</li>
    <li>sbclPackages.clerk-test</li>
    <li>sbclPackages.clgplot-test</li>
    <li>sbclPackages.clipper</li>
    <li>sbclPackages.clipper-test</li>
    <li>sbclPackages.clj</li>
    <li>sbclPackages.clog</li>
    <li>sbclPackages.clog-ace</li>
    <li>sbclPackages.clog-plotly</li>
    <li>sbclPackages.clog-terminal</li>
    <li>sbclPackages.clusters-tests</li>
    <li>sbclPackages.codex</li>
    <li>sbclPackages.codex-templates</li>
    <li>sbclPackages.coleslaw-test</li>
    <li>sbclPackages.colliflower-test</li>
    <li>sbclPackages.common-doc</li>
    <li>sbclPackages.common-doc-contrib</li>
    <li>sbclPackages.common-doc-gnuplot</li>
    <li>sbclPackages.common-doc-graphviz</li>
    <li>sbclPackages.common-doc-include</li>
    <li>sbclPackages.common-doc-plantuml</li>
    <li>sbclPackages.common-doc-plump</li>
    <li>sbclPackages.common-doc-plump-test</li>
    <li>sbclPackages.common-doc-split-paragraphs</li>
    <li>sbclPackages.common-doc-test</li>
    <li>sbclPackages.common-doc-tex</li>
    <li>sbclPackages.common-html</li>
    <li>sbclPackages.common-html-test</li>
    <li>sbclPackages.commondoc-markdown</li>
    <li>sbclPackages.commondoc-markdown-test</li>
    <li>sbclPackages.configuration_dot_options-and-quri</li>
    <li>sbclPackages.core-reader_dot_test</li>
    <li>sbclPackages.cricket_dot_test</li>
    <li>sbclPackages.cxx-test</li>
    <li>sbclPackages.datafly-test</li>
    <li>sbclPackages.dct-test</li>
    <li>sbclPackages.defclass-std-test</li>
    <li>sbclPackages.defrest</li>
    <li>sbclPackages.defrest_dot_test</li>
    <li>sbclPackages.dexador</li>
    <li>sbclPackages.dexador-test</li>
    <li>sbclPackages.dotenv-test</li>
    <li>sbclPackages.duologue</li>
    <li>sbclPackages.duologue-readline</li>
    <li>sbclPackages.duologue-test</li>
    <li>sbclPackages.eazy-documentation</li>
    <li>sbclPackages.edit-distance-test</li>
    <li>sbclPackages.elb-log</li>
    <li>sbclPackages.elb-log-test</li>
    <li>sbclPackages.envy-test</li>
    <li>sbclPackages.event-emitter-test</li>
    <li>sbclPackages.example-bot</li>
    <li>sbclPackages.fast-http-test</li>
    <li>sbclPackages.fast-websocket-test</li>
    <li>sbclPackages.freesound</li>
    <li>sbclPackages.fxml</li>
    <li>sbclPackages.gfxmath_dot_test</li>
    <li>sbclPackages.github-api-cl</li>
    <li>sbclPackages.github-gist-api-cl</li>
    <li>sbclPackages.glacier</li>
    <li>sbclPackages.glisph-test</li>
    <li>sbclPackages.hash-table-ext_dot_test</li>
    <li>sbclPackages.hello-builder</li>
    <li>sbclPackages.hello-clog</li>
    <li>sbclPackages.history-tree</li>
    <li>sbclPackages.house</li>
    <li>sbclPackages.http-body</li>
    <li>sbclPackages.http-body-test</li>
    <li>sbclPackages.ia-hash-table_dot_test</li>
    <li>sbclPackages.id3v2-test</li>
    <li>sbclPackages.inner-conditional-test</li>
    <li>sbclPackages.inquisitor-flexi-test</li>
    <li>sbclPackages.inquisitor-test</li>
    <li>sbclPackages.integral-rest</li>
    <li>sbclPackages.integral-rest-test</li>
    <li>sbclPackages.integral-test</li>
    <li>sbclPackages.intel-hex-test</li>
    <li>sbclPackages.ip-interfaces-test</li>
    <li>sbclPackages.jingle</li>
    <li>sbclPackages.jingle_dot_demo</li>
    <li>sbclPackages.jingle_dot_demo_dot_test</li>
    <li>sbclPackages.jingle_dot_test</li>
    <li>sbclPackages.jingoh</li>
    <li>sbclPackages.jingoh_dot_documentizer_dot_test</li>
    <li>sbclPackages.jingoh_dot_examiner</li>
    <li>sbclPackages.jingoh_dot_examiner_dot_test</li>
    <li>sbclPackages.jingoh_dot_generator_dot_test</li>
    <li>sbclPackages.jingoh_dot_org_dot_test</li>
    <li>sbclPackages.jingoh_dot_parallel</li>
    <li>sbclPackages.jingoh_dot_parallel_dot_test</li>
    <li>sbclPackages.jingoh_dot_reader</li>
    <li>sbclPackages.jingoh_dot_reader_dot_test</li>
    <li>sbclPackages.jingoh_dot_tester</li>
    <li>sbclPackages.jingoh_dot_tester_dot_test</li>
    <li>sbclPackages.jonathan-test</li>
    <li>sbclPackages.json-schema</li>
    <li>sbclPackages.kebab-test</li>
    <li>sbclPackages.lack-app-directory</li>
    <li>sbclPackages.lack-middleware-csrf</li>
    <li>sbclPackages.lack-middleware-session</li>
    <li>sbclPackages.lack-request</li>
    <li>sbclPackages.lack-response</li>
    <li>sbclPackages.lack-session-store-dbi</li>
    <li>sbclPackages.lack-session-store-redis</li>
    <li>sbclPackages.lack-test</li>
    <li>sbclPackages.lake-test</li>
    <li>sbclPackages.latter-day-paypal</li>
    <li>sbclPackages.legion-test</li>
    <li>sbclPackages.let-over-lambda-test</li>
    <li>sbclPackages.lisp-pay</li>
    <li>sbclPackages.lispcord</li>
    <li>sbclPackages.listopia-bench</li>
    <li>sbclPackages.lunamech-matrix-api</li>
    <li>sbclPackages.mailgun</li>
    <li>sbclPackages.marching-cubes-test</li>
    <li>sbclPackages.matrix-case_dot_test</li>
    <li>sbclPackages.mcase_dot_test</li>
    <li>sbclPackages.millet_dot_test</li>
    <li>sbclPackages.mito-attachment</li>
    <li>sbclPackages.mito-test</li>
    <li>sbclPackages.modest-config-test</li>
    <li>sbclPackages.mp3-duration-test</li>
    <li>sbclPackages.multiposter-studio</li>
    <li>sbclPackages.multiposter-tumblr</li>
    <li>sbclPackages.multival-plist-test</li>
    <li>sbclPackages.myway</li>
    <li>sbclPackages.myway-test</li>
    <li>sbclPackages.nasdf</li>
    <li>sbclPackages.nclasses</li>
    <li>sbclPackages.nfiles</li>
    <li>sbclPackages.ningle</li>
    <li>sbclPackages.ningle-test</li>
    <li>sbclPackages.njson</li>
    <li>sbclPackages.nodgui</li>
    <li>sbclPackages.north-dexador</li>
    <li>sbclPackages.nsort</li>
    <li>sbclPackages.nsymbols</li>
    <li>sbclPackages.null-package_dot_test</li>
    <li>sbclPackages.oclcl-test</li>
    <li>sbclPackages.openid-key</li>
    <li>sbclPackages.openid-key-test</li>
    <li>sbclPackages.oxenfurt</li>
    <li>sbclPackages.oxenfurt-dexador</li>
    <li>sbclPackages.pandocl</li>
    <li>sbclPackages.parenml</li>
    <li>sbclPackages.parenml-test</li>
    <li>sbclPackages.path-string-test</li>
    <li>sbclPackages.perlre</li>
    <li>sbclPackages.phos</li>
    <li>sbclPackages.poler-test</li>
    <li>sbclPackages.ppath-test</li>
    <li>sbclPackages.proc-parse-test</li>
    <li>sbclPackages.prometheus_dot_collectors_dot_process_dot_test</li>
    <li>sbclPackages.prometheus_dot_collectors_dot_sbcl_dot_test</li>
    <li>sbclPackages.prometheus_dot_exposers_dot_hunchentoot_dot_test</li>
    <li>sbclPackages.prometheus_dot_formats_dot_text_dot_test</li>
    <li>sbclPackages.prometheus_dot_pushgateway_dot_test</li>
    <li>sbclPackages.prometheus_dot_test</li>
    <li>sbclPackages.prometheus_dot_test_dot_support</li>
    <li>sbclPackages.prompt-for_dot_test</li>
    <li>sbclPackages.prompter</li>
    <li>sbclPackages.prove</li>
    <li>sbclPackages.prove-test</li>
    <li>sbclPackages.psychiq-test</li>
    <li>sbclPackages.quadtree-test</li>
    <li>sbclPackages.quantile-estimator_dot_test</li>
    <li>sbclPackages.query-repl_dot_test</li>
    <li>sbclPackages.quickutil-server</li>
    <li>sbclPackages.quickutil-utilities-test</li>
    <li>sbclPackages.quri</li>
    <li>sbclPackages.quri-test</li>
    <li>sbclPackages.read-as-string_dot_test</li>
    <li>sbclPackages.replic</li>
    <li>sbclPackages.replic-test</li>
    <li>sbclPackages.resignal-bind_dot_test</li>
    <li>sbclPackages.restful-test</li>
    <li>sbclPackages.sanity-clause</li>
    <li>sbclPackages.scriba</li>
    <li>sbclPackages.scriba-test</li>
    <li>sbclPackages.sendgrid</li>
    <li>sbclPackages.shelly-test</li>
    <li>sbclPackages.simple-config-test</li>
    <li>sbclPackages.simple-currency</li>
    <li>sbclPackages.slack-client</li>
    <li>sbclPackages.slack-client-test</li>
    <li>sbclPackages.smart-buffer-test</li>
    <li>sbclPackages.snooze</li>
    <li>sbclPackages.snooze-demo</li>
    <li>sbclPackages.snooze-tests</li>
    <li>sbclPackages.stepster</li>
    <li>sbclPackages.str_dot_test</li>
    <li>sbclPackages.stripe</li>
    <li>sbclPackages.stripe-against-the-modern-world</li>
    <li>sbclPackages.structure-ext_dot_as-class_dot_test</li>
    <li>sbclPackages.structure-ext_dot_left-arrow-accessors_dot_test</li>
    <li>sbclPackages.structure-ext_dot_make-instance_dot_test</li>
    <li>sbclPackages.sxql-test</li>
    <li>sbclPackages.t-clack-handler-fcgi</li>
    <li>sbclPackages.t-clack-handler-hunchentoot</li>
    <li>sbclPackages.t-clack-handler-toot</li>
    <li>sbclPackages.t-clack-handler-wookie</li>
    <li>sbclPackages.t-lack</li>
    <li>sbclPackages.t-lack-component</li>
    <li>sbclPackages.t-lack-middleware-accesslog</li>
    <li>sbclPackages.t-lack-middleware-auth-basic</li>
    <li>sbclPackages.t-lack-middleware-backtrace</li>
    <li>sbclPackages.t-lack-middleware-csrf</li>
    <li>sbclPackages.t-lack-middleware-mount</li>
    <li>sbclPackages.t-lack-middleware-session</li>
    <li>sbclPackages.t-lack-middleware-static</li>
    <li>sbclPackages.t-lack-request</li>
    <li>sbclPackages.t-lack-session-store-dbi</li>
    <li>sbclPackages.t-lack-session-store-redis</li>
    <li>sbclPackages.t-lack-util</li>
    <li>sbclPackages.test-gadgets</li>
    <li>sbclPackages.test-utils</li>
    <li>sbclPackages.thorn</li>
    <li>sbclPackages.thorn-doc</li>
    <li>sbclPackages.thorn-test</li>
    <li>sbclPackages.thread_dot_comm_dot_rendezvous_dot_test</li>
    <li>sbclPackages.torrents</li>
    <li>sbclPackages.torrents-test</li>
    <li>sbclPackages.translate-client</li>
    <li>sbclPackages.trestrul_dot_test</li>
    <li>sbclPackages.trivial-left-pad-test</li>
    <li>sbclPackages.trivial-string-template-test</li>
    <li>sbclPackages.trivial-ws-client</li>
    <li>sbclPackages.trivial-ws-test</li>
    <li>sbclPackages.ufo-test</li>
    <li>sbclPackages.vellum-csv-tests</li>
    <li>sbclPackages.vellum-tests</li>
    <li>sbclPackages.veq</li>
    <li>sbclPackages.vertex</li>
    <li>sbclPackages.vertex-test</li>
    <li>sbclPackages.vex</li>
    <li>sbclPackages.vivid-colors</li>
    <li>sbclPackages.vivid-colors_dot_content</li>
    <li>sbclPackages.vivid-colors_dot_content_dot_test</li>
    <li>sbclPackages.vivid-colors_dot_dispatch_dot_test</li>
    <li>sbclPackages.vivid-colors_dot_queue_dot_test</li>
    <li>sbclPackages.vivid-colors_dot_shared_dot_test</li>
    <li>sbclPackages.vivid-colors_dot_stream</li>
    <li>sbclPackages.vivid-colors_dot_stream_dot_test</li>
    <li>sbclPackages.vivid-colors_dot_test</li>
    <li>sbclPackages.vivid-diff</li>
    <li>sbclPackages.vivid-diff_dot_test</li>
    <li>sbclPackages.wallstreetflets</li>
    <li>sbclPackages.webapi</li>
    <li>sbclPackages.websocket-driver</li>
    <li>sbclPackages.websocket-driver-client</li>
    <li>sbclPackages.woo</li>
    <li>sbclPackages.woo-test</li>
    <li>sbclPackages.wookie</li>
    <li>sbclPackages.xsubseq-test</li>
    <li>sbclPackages.zacl</li>
    <li>sbclPackages.zenekindarl-test</li>
  </ul>
</details>

`sbclPackages.nyxt` fails because of an heap exhaustion, but `nyxt` succeeds since it uses `wrapLisp` with `--dynamic-memory-size 2048`, It shouldn't really matter since I don't think there's a reason to get Nyxt from `sbclPackages` rather than the `nyxt` package since it's an application and not a library.
Also, I'm not sure why it tries to build this many packages is it because of my use of `wrapLisp` or is it just how `nixpkgs-review` interacts with changes to `lisp-modules/packages.nix`?

Uthar commented:
I also got it to work by registering systems as [immutable](https://asdf.common-lisp.dev/asdf.html#index-register_002dimmutable_002dsystem-1), I picked this up from [upstream](https://github.com/atlas-engineer/nyxt/blob/c8183c0d8a0a534ae3f71f2ed95ccb21f5f3d11d/libraries/nasdf/systems.lisp#L26-L37):


```patch
--- a/pkgs/development/lisp-modules/packages.nix
+++ b/pkgs/development/lisp-modules/packages.nix
@@ -430,13 +430,10 @@ let
       pkgs.gnome.adwaita-icon-theme
     ];
 
-    postConfigure = ''
-      export CL_SOURCE_REGISTRY=$CL_SOURCE_REGISTRY:$(pwd)//
-    '';
-
     buildScript = pkgs.writeText "build-nyxt.lisp" ''
       (load "${super.nyxt.asdfFasl}/asdf.${super.nyxt.faslExt}")
       (asdf:load-system :nyxt/gi-gtk-application)
+      (map () 'asdf:register-immutable-system (asdf:already-loaded-systems))
       (sb-ext:save-lisp-and-die "nyxt" :executable t
                                        #+sb-core-compression :compression
                                        #+sb-core-compression t

```

I don't know which approach is better though, appending to CL_SOURCE_REGISTRY or registering systems as immutable. Probably we should use whichever and we'll see if issues come up.

Uthar commented:
Hmm right, so with the flag globally the dependencies would be "shared" between all lisp packages and nyxt. I would vote for increasing the heap globally, in `all-packages.nix`.

dariof4 commented:
> I also got it to work by registering systems as [immutable](https://asdf.common-lisp.dev/asdf.html#index-register_002dimmutable_002dsystem-1), I picked this up from [upstream](https://github.com/atlas-engineer/nyxt/blob/c8183c0d8a0a534ae3f71f2ed95ccb21f5f3d11d/libraries/nasdf/systems.lisp#L26-L37):
> 
> ```diff
> --- a/pkgs/development/lisp-modules/packages.nix
> +++ b/pkgs/development/lisp-modules/packages.nix
> @@ -430,13 +430,10 @@ let
>        pkgs.gnome.adwaita-icon-theme
>      ];
>  
> -    postConfigure = ''
> -      export CL_SOURCE_REGISTRY=$CL_SOURCE_REGISTRY:$(pwd)//
> -    '';
> -
>      buildScript = pkgs.writeText "build-nyxt.lisp" ''
>        (load "${super.nyxt.asdfFasl}/asdf.${super.nyxt.faslExt}")
>        (asdf:load-system :nyxt/gi-gtk-application)
> +      (map () 'asdf:register-immutable-system (asdf:already-loaded-systems))
>        (sb-ext:save-lisp-and-die "nyxt" :executable t
>                                         #+sb-core-compression :compression
>                                         #+sb-core-compression t
> ```
> 
> I don't know which approach is better though, appending to CL_SOURCE_REGISTRY or registering systems as immutable. Probably we should use whichever and we'll see if issues come up.

Nice catch! I think using `(map () 'asdf:register-immutable-system (asdf:already-loaded-systems))` should be fine since that's what upstream uses.
I did some further research and using `asdf:make` (which under the hood is just `asdf:operate :program-op`, it's also what the upstream makefile uses) would also work since it would apply `(map () 'asdf:register-immutable-system (asdf:already-loaded-systems))` thanks to the upstream function, but unfortunately I learned that if a system defines `:build-operation` (like Nyxt) then output translations are ignored for that operation (in this case `:program-op`) and it tries to place the executable where the .asd file is located, which is in the nix store, so it fails, but if `:build-operation` is not set then `asdf:operate :program-op` works normally and respects output translations, which would make it correctly place the executable in `/build/source` and make the build succeed, so that's a bit annoying :/

> Hmm right, so with the flag globally the dependencies would be "shared" between all lisp packages and nyxt. I would vote for increasing the heap globally, in all-packages.nix.

That's how I understood it working, so upping the heap globally would make sense to stop build duplication.

7c6f434c commented:
Does this globally pin quri version in sbcl.pkgs even after we update from quicklisp?

Uthar commented:
> Does this globally pin quri version in sbcl.pkgs even after we update from quicklisp?

Yes

7c6f434c commented:
Maybe we should give an attribute name with version included then?

dariof4 commented:
> Does this globally pin quri version in sbcl.pkgs even after we update from quicklisp?

Since Nyxt uses git submodules to manage their dependencies I went and updated all dependencies that were outdated in quicklisp to the versions pinned by Nyxt, some dependencies may still work if we use their quicklisp version, but to be safe I updated all of the ones that were outdated.
Once quicklisp updates it would be possible to remove a lot of manual updates for the dependencies, but if Nyxt decides to update their submodules in a later version we may run into the same issue of outdated quicklisp dependecies :/
I'm not sure if this is best solution, but I wasn't sure of any other way of doing this.

dariof4 commented:
> Hmm right, so with the flag globally the dependencies would be "shared" between all lisp packages and nyxt. I would vote for increasing the heap globally, in `all-packages.nix`.

To how much should it be increased?
Doing some testing I found out that Nyxt fails to compile with anything under ~1550MB of dynamic memory size and magicl fails with fewer than ~2300MB of heap size.

Vonfry commented:
> Since Nyxt uses git submodules to manage their dependencies I went and updated all dependencies that were outdated in quicklisp to the versions pinned by Nyxt, some dependencies may still work if we use their quicklisp version, but to be safe I updated all of the ones that were outdated. Once quicklisp updates it would be possible to remove a lot of manual updates for the dependencies, but if Nyxt decides to update their submodules in a later version we may run into the same issue of outdated quicklisp dependecies :/ I'm not sure if this is best solution, but I wasn't sure of any other way of doing this.

Could we use submodules instead of the dependencies from quicklisp or nix packaged ones? Or, we keep two version of nyxt. One is from quicklisp, and the other is manual packaged.

bhankas commented:
Really good work, @dariof4, it is truly appreciated.

I apologise for being absent on both maintenance front and this PR, but life has been occupying.

Either ways, I've not been using Nyxt for a while now, and don't consider myself qualified enough to maintain nor review. If possible could you please also remove me from the package maintainers? (old name payas, as is present in maintainers, user id is same), TIA!

I will make sure to give the new Nyxt a spin once it reached unstable, quite excited for changes, but my current machine had quite a few troubles in the past so had to stop.

dariof4 committed:
maintainers: add dariof4

7c6f434c commented:
Silly question: how much RAM to load both Nyxt _and_ magicl? (I know the compiler is hungry and so the question is whether it is «max» or «sum» in practice).

dariof4 commented:
> Could we use submodules instead of the dependencies from quicklisp or nix packaged ones? Or, we keep two version of nyxt. One is from quicklisp, and the other is manual packaged.

Using submodules instead of quicklisp dependecies is possible, although that would cause some extra building due to not sharing Nyxt's dependecies with other lisp packages.

> Really good work, @dariof4, it is truly appreciated.
> 
> I apologise for being absent on both maintenance front and this PR, but life has been occupying.
> 
> Either ways, I've not been using Nyxt for a while now, and don't consider myself qualified enough to maintain nor review. If possible could you please also remove me from the package maintainers? (old name payas, as is present in maintainers, user id is same), TIA!
> 
> I will make sure to give the new Nyxt a spin once it reached unstable, quite excited for changes, but my current machine had quite a few troubles in the past so had to stop.

Okay, I'll remove you from the Nyxt maintainers. Thanks for your previous work maintaining it!

> Silly question: how much RAM to load both Nyxt _and_ magicl? (I know the compiler is hungry and so the question is whether it is «max» or «sum» in practice).

I'm not exactly sure how to test that since trying to build Nyxt and magicl simultaneously (`nix build .#sbclPackges.{nyxt, magicl}`) seems to always fail, I think it's because `sb-ext:save-lisp-and-die` kills the `sbcl` process, since this happens with stumpwm as well.

patrickaldis commented:
I was just playing around with this PR, and everything runs great, but I realised that I get a lot of warnings in the console about portals:
```
(WebKitWebProcess:2): Gdk-WARNING **: 10:13:29.316: Failed to read portal settings: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.Settings” on object at path /org/freedesktop/portal/desktop
```
And noticed that the portal location is pointing to `/org/freedeskop/portal/desktop` rather than a store location.

Is this an issue on my side that I don't have the right desktop portal installed, or should this point to a store path?

Thanks for all the great work!

dariof4 committed:
sbcl: add more heap size

Set the `--dynamic-spzce-size` flag for sbcl to add more heap size to
make certain packages be able to compile.
ref: https://github.com/NixOS/nixpkgs/pull/235095#issuecomment-1574352970

dariof4 committed:
lisp-modules.nyxt-gtk: use gi-gtk system instead of gtk system

Package :nyxt/gi-gtk-application instead of :nyxt/gtk-application.
ref: https://github.com/NixOS/nixpkgs/issues/217888#issuecomment-1562523130

dariof4 commented:
> I was just playing around with this PR, and everything runs great, but I realised that I get a lot of warnings in the console about portals:
> 
> ```
> (WebKitWebProcess:2): Gdk-WARNING **: 10:13:29.316: Failed to read portal settings: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.Settings” on object at path /org/freedesktop/portal/desktop
> ```
> 
> And noticed that the portal location is pointing to `/org/freedeskop/portal/desktop` rather than a store location. You should be able to reproduce this by fullscreening a video, the browser should attempt to use portals to inhibit sleep.
> 
> Is this an issue on my side that I don't have the right desktop portal installed, or should this point to a store path?
> 
> Thanks for all the great work!

I unfortunale can't reproduce :(, are you using wayland or x11? And what gpu drivers are you using?
In the meantime I found out about a few other issues:

1. Copy/Paste doesn't work since `(uiop:run-program "PROGRAM" :output :string)` tries to create `/build` for some reason, unless Nyxt is compiled with `asdf:make` or `asdf:operate :program-op` :shrug: I also had to add `xclip` to `PATH` with `makeWrapper`.

2. If you're using nvidia drivers videos don't work unless `WEBKIT_DISABLE_COMPOSITING_MODE` is set to `1`, this seems to be a webkitgtk issues though since ephiphany has the same problem.
see #221102 and #32580

I updated the branch with the fix for copy/pasting, and I set the `--dynamic-space-size` flag for sbcl.

7c6f434c commented:
I think maintainer GitHub id needs to be in the same representation as the other ids in the file

dariof4 commented:
> I think maintainer GitHub id needs to be in the same representation as the other ids in the file

Oops, that was a dumb mistake :sweat_smile:, I fixed it.

7c6f434c commented:
I still think that if we have quicklisp mass-import, then version overrides are better with versions in the attribute names.

(I am not 100% sure what is the proper way to handle Nyxt so that «you can't load Nyxt and magicl at once _at all_» goes away, but that question is out of scope…)

dariof4 commented:
> I still think that if we have quicklisp mass-import, then version overrides are better with versions in the attribute names.
> 
> (I am not 100% sure what is the proper way to handle Nyxt so that «you can't load Nyxt and magicl at once _at all_» goes away, but that question is out of scope…)

I went ahead and switched the pr to naming the attributes rather than shadowing them, this required explicitly listing all the `lispLibs` for Nyxt since if I tried to keep the ones from quicklisp it would prefer quicklisp version of the libraries rather than the updated ones.
